cprover_bindings/src/goto_program/typ.rs

```diff
@@ -570,6 +570,87 @@ impl Type {
         }
     }
 
+    /// A transparent type wraps a base type inside a struct, and has the same in-memory layout.
+    /// For example,
+    /// ```rust
+    ///     struct TransparentWrapper { int x };
+    /// ```
+    /// We define it recursively as a type which has exactly one field, which it itself either
+    /// a transparent type, or is a scalar type.
+    pub fn is_transparent_type(&self, st: &SymbolTable) -> bool {
+        match self {
+            // Follow tags to get the underlying structure
+            StructTag(tag) | UnionTag(tag) => st.lookup(*tag).unwrap().typ.is_transparent_type(st),
+
+            // Recursively check: does this only have one field, which is either a wrapper or base type.
+            Struct { .. } | Union { .. } => self.unwrap_transparent_type(st).is_some(),
+
+            // Base types
+            Array { .. }
+            | Bool
+            | CBitField { .. }
+            | CInteger(_)
+            | Code { .. }
+            | Constructor
+            | Double
+            | Empty
+            | FlexibleArray { .. }
+            | Float
+            | IncompleteStruct { .. }
+            | IncompleteUnion { .. }
+            | InfiniteArray { .. }
+            | Pointer { .. }
+            | Signedbv { .. }
+            | Unsignedbv { .. }
+            | VariadicCode { .. }
+            | Vector { .. } => false,
+        }
+    }
+
+    /// Given a transparent type (see comment on `Type::is_transparent_type()`),
+    /// extract the type it wraps.
+    pub fn unwrap_transparent_type(&self, st: &SymbolTable) -> Option<Type> {
+        match self {
+            Array { .. }
+            | Code { .. }
+            | Constructor
+            | FlexibleArray { .. }
+            | IncompleteStruct { .. }
+            | IncompleteUnion { .. }
+            | InfiniteArray { .. }
+            | VariadicCode { .. }
+            | Vector { .. } => None,
+
+            // Base types
+            Bool
+            | CBitField { .. }
+            | CInteger(_)
+            | Double
+            | Empty
+            | Float
+            | Pointer { .. }
+            | Signedbv { .. }
+            | Unsignedbv { .. } => Some(self.clone()),
+
+            // Follow tags to get the underlying structure
+            StructTag(tag) | UnionTag(tag) => {
+                st.lookup(*tag).unwrap().typ.unwrap_transparent_type(st)
+            }
+
+            // Recursively check: does this only have one field, which is either a wrapper or base type.
+            Struct { components, .. } | Union { components, .. } => {
+                if components.len() != 1 {
```

Comment:
Can you please check if that's the case for transparent types that have extra zero-sized fields as described here? https://doc.rust-lang.org/1.26.2/unstable-book/language-features/repr-transparent.html#enter-reprtransparent

Reply:
Added a TODO and a tracking ticket